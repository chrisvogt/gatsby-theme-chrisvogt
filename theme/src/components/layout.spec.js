import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeUIProvider } from 'theme-ui'

import Footer from './footer'
import Layout from './layout'
import TopNavigation from './top-navigation'

// Mock Footer and TopNavigation components
jest.mock('./footer')
jest.mock('./top-navigation')

describe('Layout', () => {
  TopNavigation.mockImplementation(() => <div className='MOCK__TopNavigation'></div>)
  Footer.mockImplementation(() => <div className='MOCK__Footer'></div>)

  // Mock theme
  const mockTheme = {
    colors: {
      background: '#fdf8f5',
      text: '#111',
      modes: {
        dark: {
          background: '#1e1e2f',
          text: '#fff'
        }
      }
    }
  }

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <ThemeUIProvider theme={mockTheme}>
          <Layout>
            <div className='fake-website'>
              <h1>Fake Website</h1>
              <p>Lorum ipsum dolor sit amet.</p>
            </div>
          </Layout>
        </ThemeUIProvider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
