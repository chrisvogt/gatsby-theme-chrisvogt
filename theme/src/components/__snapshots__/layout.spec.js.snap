// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches the snapshot 1`] = `
[
  <a
    data-reach-skip-link=""
    data-reach-skip-nav-link=""
    href="#reach-skip-nav"
  >
    Skip to content
  </a>,
  <div
    className="css-tvtok6-Layout"
  >
    <div
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <canvas
        style={
          {
            "display": "block",
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
      <div
        style={
          {
            "WebkitBackdropFilter": "blur(100px)",
            "backdropFilter": "blur(100px)",
            "backgroundColor": "rgba(75, 0, 130, 0.02)",
            "height": "100%",
            "left": 0,
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
    <header
      className="css-1jqf6qp-Layout"
      role="banner"
    >
      <div
        className="MOCK__TopNavigation"
      />
    </header>
    <main
      role="main"
    >
      <div
        data-reach-skip-nav-content=""
        id="reach-skip-nav"
      />
      <div
        className="fake-website"
      >
        <h1>
          Fake Website
        </h1>
        <p>
          Lorum ipsum dolor sit amet.
        </p>
      </div>
    </main>
    <div
      className="MOCK__Footer"
    />
  </div>,
]
`;
